torch/nn/modules/transformer.py

```diff
@@ -0,0 +1,519 @@
+        # create two learnable parameters to calibrate normalisation
+        self.alpha = Parameter(torch.ones(self.size))
+        self.bias = Parameter(torch.zeros(self.size))
+        self.eps = eps
+
+    def forward(self, x):
+        r"""Inputs of forward function
+
+        Args:
+            x: the tensor fed to the LayerNorm model (required).
+
+        Shape:
+            x: [sequence length, batch size, embed dim]
+
+            output: [sequence length, batch size, embed dim]    
+
+        Examples:
+            >>> output = layer_norm(x)
+        """
+        x_mean = x.mean(dim=-1, keepdim=True)
+        x_std = x.std(dim=-1, keepdim=True)
+        return self.alpha * (x - x_mean) / (x_std + self.eps) + self.bias
+
+
+# Temporarily leave FeedForward module here. Will be moved somewhere else.
+class FeedForward(Module):
+    r"""A fully connected feed-forward network, which consists of two 
+        linear transformations with a ReLU activation in between.
+
+    Args:
+        d_model: the embed dim (required).
+        d_ff: the dimension of the feedforward network model (default=2048).
+        dropout: the dropout value (default=0.1).
+
+    Examples::
+        >>> ff = FeedForward(d_model)
+    """
+
+    def __init__(self, d_model, d_ff=2048, dropout=0.1):
+        super(FeedForward, self).__init__() 
+        self.linear1 = Linear(d_model, d_ff)
+        self.dropout = Dropout(dropout)
+        self.linear2 = Linear(d_ff, d_model)
+
+    def forward(self, x):
+        r"""Inputs of forward function
+
+        Args:
+            x: the tensor fed to FeedForward model (required).
+
+        Shape:
+            x: [sequence length, batch size, embed dim]
+
+            output: [sequence length, batch size, embed dim]    
+
+        Examples:
+            >>> output = ff(x)
+        """
+        return self.linear2(self.dropout(F.relu(self.linear1(x))))
+
+
+# Temporarily leave Embeddings module here. Will be moved somewhere else.
+class Embeddings(Module):
+    r"""Sequence embedding. Normalized by math.sqrt(embed dim).
+
+    Args:
+        d_model: the embed dim (required).
+        vocab: the number of vocabularies in the sequence (required). 
+
+    Examples:
+        >>> embed_model = Embeddings(d_model, vocab)
+    """
+    def __init__(self, d_model, vocab):
+        super(Embeddings, self).__init__()
+        self.lut = Embedding(vocab, d_model)
+        self.d_model = d_model
+
+    def forward(self, x):
+        r"""Inputs of forward function
+
+        Args:
+            x: the sequence fed to the embedding model (required).
+
+        Shape:
+            x: [sequence length, batch size]
+
+            output: [sequence length, batch size, embed dim]    
+
+        Examples:
+            >>> output = embed_model(x)
+        """
+
+        return self.lut(x) * math.sqrt(self.d_model)
+
+
+# Temporarily leave PositionalEncoding module here. Will be moved somewhere else.
+class PositionalEncoding(Module):
```

Comment:
This should be left to the user. It's something you could define locally for your integration test.